### ISSUE TITLE
Spinner does not clear the line on update

### ISSUE THREAD

**Issue by tzakharko** [r-lib/cli#162]
When a spinner is updated, it does not clear the line properly. This can produce artifacts when the template is changed on the fly (e.g. when one wants to produce a different into string while the spinner is running). 

Suggestion to fix: change

https://github.com/r-lib/cli/blob/e3ca34656f5bb82df63bfc1c741e75acc79b13d9/R/spinner.R#L165

to

~~~r
    clear_line()
    cat(line, sep = "", file = stream) 
~~~

Not submitting a pull request since the fix is this trivial...

**Comment by AkhilGNair** [comment]
Specifically I think OP is referring to when the template changes length so the line isn't fully overwritten. Might be wrong though.

```
x = 1
spinner = cli::make_spinner()

while (x < 100) {
  if (x %% 3 == 0) { template = "looooong" } else { template = "short" }
  spinner$spin(glue::glue("{{spin}} {template}"))
  x = x + 1
  Sys.sleep(0.05)
}
spinner$finish()
```

Occasionally this will print

```
⠧ shortong
```

**Comment by tzakharko** [comment]
@AkhilGNair yes, you are entirely correct, this is the problem. @gaborcsardi mentions that just adding `clear_line()` will cause flickering, I suppose it is an issue with some terminal emulators (I have not seen any flickering using tools I am working with). 

I have now submitted a fix that caches the width of the previously displayed line, adding padding to clear out the garbage if the template becomes shorter

**Comment by gaborcsardi** [comment]
Fixed by #164.
